### PR TITLE
Simplify substring() usage in MimeTypeUtils

### PR DESCRIPTION
**Pull request by mikaelparsekyan**
Removed redundant if-constructions and unnecessary variable declaration.

### CODE REVIEW

File: spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java [hunk 1]
```diff
@@ -569,8 +569,7 @@ private Object createHelper() {
 				interceptDuringConstruction,
 				serialVersionUID);
 		this.currentKey = key;
-		Object result = super.create(key);
-		return result;
+		return super.create(key);
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java:572]: Please revert this according to our policy:

> Please refrain from modifying classes under `org.springframework.asm`, `org.springframework.cglib`, and `org.springframework.objenesis`. Those include repackaged forks of the third-party libraries ASM, CGLIB, and Objenesis. Any refactoring to those classes should take place upstream in the originating repository. The Spring Framework will then pick up the changes when syncing with official updates of the forked third-party libraries.

File: spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java [hunk 2]
```diff
@@ -826,8 +825,7 @@ protected Object wrapCachedClass(Class klass) {
 	@Override
 	protected Object unwrapCachedValue(Object cached) {
 		if (currentKey instanceof EnhancerKey) {
-			EnhancerFactoryData data = ((WeakReference<EnhancerFactoryData>) cached).get();
-			return data;
+			return ((WeakReference<EnhancerFactoryData>) cached).get();
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java:828]: Please revert this according to our policy:

> Please refrain from modifying classes under `org.springframework.asm`, `org.springframework.cglib`, and `org.springframework.objenesis`. Those include repackaged forks of the third-party libraries ASM, CGLIB, and Objenesis. Any refactoring to those classes should take place upstream in the originating repository. The Spring Framework will then pick up the changes when syncing with official updates of the forked third-party libraries.

File: spring-core/src/main/java/org/springframework/core/annotation/AnnotationUtils.java [hunk 3]
```diff
@@ -164,9 +164,6 @@ public static boolean isCandidateClass(Class<?> clazz, Class<? extends Annotatio
 	 * @see #isCandidateClass(Class, Class)
 	 */
 	public static boolean isCandidateClass(Class<?> clazz, String annotationName) {
-		if (annotationName.startsWith("java.")) {
-			return true;
-		}
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/core/annotation/AnnotationUtils.java:169]: This check is not redundant.

This checks the annotation type; whereas, the subsequent `hasPlainJavaAnnotationsOnly()` invocation checks the candidate class.

Thus, please revert this change.

File: spring-core/src/main/java/org/springframework/util/MimeTypeUtils.java [hunk 4]
```diff
@@ -219,7 +219,7 @@ private static MimeType parseMimeTypeInternal(String mimeType) {
 			throw new InvalidMimeTypeException(mimeType, "does not contain subtype after '/'");
 		}
 		String type = fullType.substring(0, subIndex);
-		String subtype = fullType.substring(subIndex + 1, fullType.length());
+		String subtype = fullType.substring(subIndex + 1);
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/util/MimeTypeUtils.java:222]: Good catch

File: spring-core/src/main/java/org/springframework/util/MimeTypeUtils.java [hunk 5]
```diff
@@ -248,7 +248,7 @@ else if (ch == '"') {
 				int eqIndex = parameter.indexOf('=');
 				if (eqIndex >= 0) {
 					String attribute = parameter.substring(0, eqIndex).trim();
-					String value = parameter.substring(eqIndex + 1, parameter.length()).trim();
+					String value = parameter.substring(eqIndex + 1).trim();
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/util/MimeTypeUtils.java:251]: ditto

**Review by sbrannen — CHANGES_REQUESTED** [review]
Thanks for the PR.

I have requested that you revert a few of the changes.

Once you've done that, we'll be able to merge this.

**Comment by mikaelparsekyan** [comment]
I did the changes.

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks

**Comment by mikaelparsekyan** [comment]
I am glad I helped.

**Pull request merged**